src/blueprints/tv2/timeline-object-factories/tv2-atem-video-mixer-timeline-object-factory.ts

```diff
@@ -66,40 +71,119 @@ export class Tv2AtemVideoMixerTimelineObjectFactory implements Tv2VideoMixerTime
     return Tv2AtemLayer.DOWNSTREAM_KEYER
   }
 
-  public createProgramTimelineObject(sourceInput: number, enable: TimelineEnable): AtemMeTimelineObject {
+  public createUpstreamKeyerTimelineObject(downstreamKeyer: Tv2DownstreamKeyer, enable: TimelineEnable): AtemMeTimelineObject {
+    const downstreamKeyerNumber: number = downstreamKeyer.Number + 1
     return {
-      id: `${ATEM_PREFIX}program`,
+      id: `${ATEM_PREFIX}upstreamKeyer${downstreamKeyerNumber}`,
       enable,
       priority: 1,
-      layer: Tv2AtemLayer.PROGRAM,
+      layer: Tv2AtemLayer.CLEAN_UPSTREAM_KEYER,
       content: {
         deviceType: DeviceType.ATEM,
         type: AtemType.ME,
         me: {
-          input: sourceInput,
-          transition: AtemTransition.CUT
+          upstreamKeyers: [
+            {
+              upstreamKeyerId: downstreamKeyer.Number,
+              onAir: true,
+              mixEffectKeyType: 0,
+              flyEnabled: false,
+              fillSource: downstreamKeyer.Fill,
+              cutSource: downstreamKeyer.Key,
+              maskEnabled: false,
+              lumaSettings: {
+                clip: this.convertPercentageToAtemPercentageValue(downstreamKeyer.Clip),
+                gain: this.convertPercentageToAtemPercentageValue(downstreamKeyer.Gain),
+              }
+            }
+          ]
         }
       }
     }
   }
 
-  public createCleanFeedTimelineObject(sourceInput: number, enable: TimelineEnable): AtemMeTimelineObject {
+  public createProgramTimelineObject(sourceInput: number, enable: TimelineEnable, transition?: {type: VideoMixerTransition, settings: VideoMixerTransitionSettings}): AtemMeTimelineObject {
+    return this.createAtemMeTimelineObjectForLayer(
+      `${ATEM_PREFIX}program`,
+      Tv2AtemLayer.PROGRAM,
+      enable,
+      {
+        input: sourceInput,
+        transition: transition ? this.mapVideoMixerTransitionToAtemVersion(transition.type) : undefined,
```

Comment:
This is a breaking change that will make Atem cut to black when `transition` is `undefined`.
To avoid merge-conflicts, I will fix this in SOF-1696

Reply:
Woops my bad. Was suppose to default to `AtemTransition.Cut`, if no transition was given.